test/specs/components/Popup/Popup-test.tsx

```diff
@@ -36,6 +38,57 @@ describe('Popup', () => {
   }: PositionTestInput & { rtl?: never }) =>
     testPopupPosition({ align, position, expectedPlacement, rtl: true })
 
+  const getPopupContent = (popup: ReactWrapper) => {
+    return popup.find('.ui-popup__content')
+  }
+
+  const togglePopupTest = (keyName: string, keyboardKey: keyboardKey, onProp: PopupEvents) => {
+    test(`toggle popup with ${keyName} key`, () => {
+      const triggerId = 'triggerElement'
+      const popupContent = 'any test content'
+      const popup = mountWithProvider(
+        <Popup
+          trigger={<span id={triggerId}> text to trigger popup </span>}
+          content={popupContent}
+          on={onProp}
+        />,
+      )
+      const popupTriggerElement = popup.find(`#${triggerId}`)
+      popupTriggerElement.simulate('keydown', { keyCode: keyboardKey })
```

Comment:
I would say that, if the span is not focusable we will not be able to repro this in a real usage. Unfortunately, no better idea at this moment..